### PR TITLE
GH1251: added channel param to CreateReleaseArgumentBuilder

### PR DESCRIPTION
**Pull request by vip32**
this is needed to create octopus releases for a specific channel

### CODE REVIEW

File: src/Cake.Common/Tools/OctopusDeploy/CreateReleaseSettings.cs [hunk 1]
```diff
@@ -46,5 +46,15 @@ public sealed class CreateReleaseSettings : OctopusDeploySettings
...
         /// </summary>
         public bool IgnoreExisting { get; set; }
+
+        /// <summary>
+        /// Gets or sets the octopuschannel for the new release.
+        /// </summary>
+        public string Channel { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether channel rules should be ignored
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/OctopusDeploy/CreateReleaseSettings.cs]: Missing period after `ignored`.

Comment (gep13) [inline src/Cake.Common/Tools/OctopusDeploy/CreateReleaseSettings.cs]: @patriksvensson shouldn't there also be a new test for checking the settings of this property?

Comment (vip32) [inline src/Cake.Common/Tools/OctopusDeploy/CreateReleaseSettings.cs]: i didn't find any tests regarding octopus settings or argumentbuilder, so was not able to add this

Comment (vip32) [inline src/Cake.Common/Tools/OctopusDeploy/CreateReleaseSettings.cs]: period is added now.

Comment (gep13) [inline src/Cake.Common/Tools/OctopusDeploy/CreateReleaseSettings.cs]: Oh really. That is something we need to correct in a separate PR then.

Comment (gep13) [inline src/Cake.Common/Tools/OctopusDeploy/CreateReleaseSettings.cs]: Wait, doesn't these tests cover this tool: https://github.com/cake-build/cake/blob/develop/src/Cake.Common.Tests/Unit/Tools/OctopusDeploy/OctoCreateReleaseTests.cs

On phone just now, so haven't checked properly.

Comment (patriksvensson) [inline src/Cake.Common/Tools/OctopusDeploy/CreateReleaseSettings.cs]: Yes, @gep13 is correct. We should add tests for this.

Comment (vip32) [inline src/Cake.Common/Tools/OctopusDeploy/CreateReleaseSettings.cs]: relevant tests are added. i also added some settings to trigger a deployment when creating an octo release (deployto/progress from http://docs.octopusdeploy.com/display/OD/Creating+releases).

Comment (vip32) [inline src/Cake.Common/Tools/OctopusDeploy/CreateReleaseSettings.cs]: is this good enough to pull now?

**Comment by gep13** [comment]
@vip32 would you be in a position to rebase this PR about the latest develop branch, and squash into a single commit?  Thanks!

**Comment by gep13** [comment]
@vip32 in addition to my above comment, was there an issue created for this problem?

As per our contributing guide:

https://github.com/cake-build/cake/blob/develop/CONTRIBUTING.md

We like to have an issue that we can reference, can you please create one?

**Comment by vip32** [comment]
issue for this PR https://github.com/cake-build/cake/issues/1251

**Comment by gep13** [comment]
@vip32 are you in a position to rebase this PR into a single commit?  You can find instructions on how to do this in the contributing.md I linked to earlier.

File: src/Cake.Common/Tools/OctopusDeploy/CreateReleaseSettings.cs (hunk 1 above)
Comment (vip32) [inline src/Cake.Common/Tools/OctopusDeploy/CreateReleaseSettings.cs]: rebased and squashed, please pull

Comment (gep13) [inline src/Cake.Common/Tools/OctopusDeploy/CreateReleaseSettings.cs]: @vip32 I am still seeing 11 commits in this PR. Can you squash them?

Comment (vip32) [inline src/Cake.Common/Tools/OctopusDeploy/CreateReleaseSettings.cs]: seems to be one commit now, a push --force was needed

**Comment by gep13** [comment]
image

**Comment by devlead** [comment]
@vip32 your changes have now been merged, thanks for your contribution 👍

**Comment by vip32** [comment]
nice, was a rough one but learned some new things too. next time i am better prepared ;)
keep up the good work!

**Pull request merged**